tools/cmake/mbed_greentea.cmake

```diff
@@ -37,42 +52,53 @@ macro(mbed_greentea_add_test)
         "${multipleValueArgs}"
         ${ARGN}
     )
-    add_subdirectory(${MBED_ROOT} build)
+    if(NOT "${MBED_GREENTEA_TEST_SKIPPED}" STREQUAL "")
+        set(msg "Skipping ${MBED_GREENTEA_TEST_NAME}: ${MBED_GREENTEA_TEST_SKIPPED}")
+        message("${msg}")
+        add_custom_target(${MBED_GREENTEA_TEST_NAME}
+            ALL
+            COMMAND ${CMAKE_COMMAND} -E echo ${msg}
+        )
```

Comment:
**For discussion**: To we want skip messages?

If we want it at when compiling tests, it also makes sense to have it for CTest running stage. To do that (assuming `add_test()` is called first),
```cmake
set_tests_properties(${MBED_GREENTEA_TEST_NAME}
    PROPERTIES DISABLED
)
```
See test properties: https://cmake.org/cmake/help/latest/manual/cmake-properties.7.html#properties-on-tests

Reply:
The idea of printing a "skip" message is to let users know skipped tests have been considered

Reply:
> **For discussion**: Do we want to print a skip message?
> 
> If we want it, it also makes sense to have its CTest test defined but skipped for consistency. To do that (assuming `add_test()` is called first),
> 
> ```cmake
> set_tests_properties(${MBED_GREENTEA_TEST_NAME}
>     PROPERTIES DISABLED
> )
> ```
> 
> See tests properties: https://cmake.org/cmake/help/latest/manual/cmake-properties.7.html#properties-on-tests
> 
> If we don't want a message, no change is needed in mbed_greentea.cmake - just don't add the test in its own CMakeLists.txt.

I think we should define the skipped test, and set the test properties to report the test as skipped in the CTest test report. This would be useful if we ever wanted to use the CTest report to populate a dashboard view of test results. I think it's more useful to show the test as skipped in the CTest test report than it is to print a "skipped" message at config time.

Reply:
I'll set the test properties when rebasing this PR once #14892 is merged.

I think it doesn't hurt to both print "skipped" at config time and having it in the CTest test report, though the config time message is a bit redundant and can be removed. And there's also a build time message in this PR to let the user know why a test is not built.

As for the precise test property to use, I found `DISABLED True` not entirely suitable because it prints

    1/1 Test #1: mbed-drivers-ticker ..............***Not Run (Disabled)  0.00 sec

and the user might ask why it's "Not Run (Disabled)". So I'm thinking of

```cmake
add_test(
    NAME ${MBED_GREENTEA_TEST_NAME}
    COMMAND ${CMAKE_COMMAND} -E echo "Skipping ${MBED_GREENTEA_TEST_NAME}"
}
set_tests_properties(${MBED_GREENTEA_TEST_NAME}
    PROPERTIES
        SKIP_REGULAR_EXPRESSION "."
)
```

which gives

    1/1 Test #1: mbed-drivers-ticker ..............***Skipped   0.01 sec